src/averageAges.js

```diff
@@ -55,7 +88,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  function findMother(child) {
+    return people.find(mother => mother.name === child.mother);
+  }
+
+  let family = people.filter(findMother);
+
+  if (onlyWithSon === true) {
+    family = family.filter(child => child.sex === 'm');
+  }
+
+  const ageDif = family.map(child => child.born - findMother(child).born);
+
+  const avrAge = ageDif.reduce((age1, age2) => age1 + age2) / family.length;
```

Comment:
it's not a `sort` method, name your variables the proper way, like `sumOfAges` and `currentAge`

Reply:
done